jdaviz/app.py

```diff
@@ -2048,15 +2047,28 @@ def _create_data_item(data):
             typ = 'cube'
         else:
             typ = 'unknown'
+
         # we'll expose any information we need here.  For "meta", not all entries are guaranteed
         # to be serializable, so we'll just send those that we need.
+
+        def _expose_meta(key):
+            """
+            whether to expose the metadata to the data-item in the UI based on the dictionary key
+            """
+            if key in ('Plugin', 'mosviz_row'):
+                return True
+            if key.lower().startswith(f'_{self.config}'):
+                # other internal metadata (like lcviz's '_LCVIZ_EPHEMERIS')
+                return True
```

Comment:
I am not comfortable with this. Anything with `_imviz*` is going to show too.

Reply:
right, is that a problem with anything?  I was trying to make this general in case it was useful elsewhere, but if we'd rather have lcviz just directly hardcoded here as well, I'm ok with that for now.

Reply:
Widely accepted Python convention is anything with leading underscore is not public. This goes against that and it is not documented in any dev docs. A different dev might accidentally stumble on this with unintended behavior?

Reply:
Why doesn't lcviz use `LCVIZ_EPHEMERIS` directly without leading underscore?

Reply:
the leading underscore was because it is not meant to be public and should be hidden from the metadata viewer plugin (its just for internal tracking of where it came from).

Reply:
So the "metadata" mentioned in "whether to expose the metadata" docstring has nothing to do with Metadata plugin? I guess I am still confused.

Reply:
Right, this is exposing some of the metdata stored with the glue data object to the data-item accessible from vue (which in turn is used to determine which to display in the data menu for each viewer).  The only thing it has to do with the metadata viewer plugin is that also reads from the same glue metadata, and already includes logic to ignore entries that start with an underscore - which I took to mean that that would be the correct place to attach internal information to the glue data object.  If there is a better place to attach this internal information, I don't oppose to moving it, we just need some way to store information in data created by plugins that is then retrievable elsewhere.

Reply:
The only possible risk of overexposing `_config*` entries, as far as I can tell, would be if they are not serializable (which would throw an error, so we would then quickly see the problem) or contain a large quantity of information that would affect performance.

Reply:
I suggested docstring changes. Maybe it was a matter to make it a little clearer.

Reply:
Okay. Clearer now. Thanks!